lib/Resque/Job.php

```diff
@@ -188,18 +191,49 @@ public function getInstance()
 	 */
 	public function perform()
 	{
+
+		$method = 'perform';
+		$beforePerformMethod = 'setUp';
+		$tearDownMethod = 'tearDown';
 		$instance = $this->getInstance();
+		if (is_array($instance))
+		{
+			if (isset($instance[4]))
```

Comment:
These are a way to allow helper class Resque_Job_Creator to be more flexible and use variable names for "perform" , "setUP" and "tearDown" methods to better match underneat job classes
For example to use CakePHP shells as jobs: https://github.com/wa0x6e/Cake-Resque/pull/92